From ed10d22c3c4005ba89152e70f83f9494a41b5a56 Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 10 Dec 2014 10:29:16 -0500
Subject: [wireless] Revert: iwlwifi/dvm: don't enable CTS to self

Message-id: <1418207389-7899-3-git-send-email-sgruszka@redhat.com>
Patchwork-id: 101574
O-Subject: [RHEL7.1 PATCH 02/35] Revert "iwlwifi: dvm: don't enable CTS to self"
Bugzilla: 1155538
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

From: Emmanuel Grumbach <emmanuel.grumbach@intel.com>

BZ#1155538

commit f47f46d7b09cf1d09e4b44b6cc4dd7d68a08028c upstream.

This reverts commit 43d826ca5979927131685cc2092c7ce862cb91cd.

This commit caused packet loss.

Signed-off-by: Emmanuel Grumbach <emmanuel.grumbach@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8330268

Testing:
Tested on iwl5100, iwl6300, iwl6205 adapters.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/wireless/iwlwifi/dvm/rxon.c b/drivers/net/wireless/iwlwifi/dvm/rxon.c
index 6dc5dd3..ed50de6 100644
--- a/drivers/net/wireless/iwlwifi/dvm/rxon.c
+++ b/drivers/net/wireless/iwlwifi/dvm/rxon.c
@@ -1068,6 +1068,13 @@ int iwlagn_commit_rxon(struct iwl_priv *priv, struct iwl_rxon_context *ctx)
  /* recalculate basic rates */
  iwl_calc_basic_rates(priv, ctx);
 
+ /*
+  * force CTS-to-self frames protection if RTS-CTS is not preferred
+  * one aggregation protection method
+  */
+ if (!priv->hw_params.use_rts_for_aggregation)
+  ctx->staging.flags |= RXON_FLG_SELF_CTS_EN;
+
  if ((ctx->vif && ctx->vif->bss_conf.use_short_slot) ||
      !(ctx->staging.flags & RXON_FLG_BAND_24G_MSK))
   ctx->staging.flags |= RXON_FLG_SHORT_SLOT_MSK;
@@ -1473,6 +1480,11 @@ void iwlagn_bss_info_changed(struct ieee80211_hw *hw,
  else
   ctx->staging.flags &= ~RXON_FLG_TGG_PROTECT_MSK;
 
+ if (bss_conf->use_cts_prot)
+  ctx->staging.flags |= RXON_FLG_SELF_CTS_EN;
+ else
+  ctx->staging.flags &= ~RXON_FLG_SELF_CTS_EN;
+
  memcpy(ctx->staging.bssid_addr, bss_conf->bssid, ETH_ALEN);
 
  if (vif->type == NL80211_IFTYPE_AP ||
-- 
1.7.1